lib/string_reverse.rb

```diff
@@ -1,4 +1,30 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  # my_string << "not implemented"
+  if my_string == nil
+    return nil
+  end
+
+  i = 0
+  j = my_string.length - 1
+  while i < j
+    first_letter = my_string[i]
+    last_letter = my_string[j]
+
+    my_string[i] = last_letter
+    my_string[j] = first_letter
+
+    i += 1
+    j -= 1
+  end
+

```

Comment:
i tried running this code, and couldn't get it to return anything, even implicitly. i added "return my_string" in this line, and it worked. maybe i copied and pasted wrong though.